### ISSUE TITLE
Import database fails to recognize file

### ISSUE THREAD

**Issue by xordspar0** [iSoron/uhabits#616]
I exported a database before switching phones. Now on my new phone, when I try to import the database Habits says "File not recognized."

Does that mean that it's detecting it as an unknown third party app export, or is it recognizing it as a Habits database but failing to import it for some reason?

**Comment by KristianTashkov** [comment]
Looking at the code it tries to import it from all possible sources and if none of them work, it returns this error message.
If you exported the file from within the app, it looks like failure modes are either if the file is corrupted in some way and has a different SQL structure than it should, or if it's exported from a newer version of the app, than the one you are trying to import into. Maybe you had the beta version from google play on your old phone but you are importing to the regular version?

**Comment by xordspar0** [comment]
~I'm using the newest version from f-droid. The database was exported from a slightly older version from f-droid.~

Nope, in the process of checking what versions I had, I realized the Google restored a version of Habits that was *older* than what I used to have. Updating to the latest fixed it.

Thanks for clarifying how the restore code works. I looked at it myself but I wasn't sure if Habits was even recognizing it as a valid database, even though I could open it with sqlite on the command line.
